src/components/App/App.js

```diff
@@ -41,26 +43,31 @@ class App extends React.Component {
     const { pokemons, error, caughtPokemon} = this.state
     return (
       <main>
-        <Navbar />
-        <Switch>
-        <Route path="/login" component={Login} />
-        <Route
-        exact path="/"
-        render={() => error
-          ? <h2>{error}</h2>
-          : <Home pokemons={pokemons} caught={caughtPokemon} favorite={this.catchPokemon} />}
-        />
-        <Route path='/caught' render={() => error
-          ? <h2>{error}</h2>
-          : <Caught pokemons={pokemons} caught={caughtPokemon} favorite={this.catchPokemon} />}
-          />
-        <Route
-        path="/:id"
-        render={({match}) => {
-          const id = match.params.id;
-          return <PokemonDetails id={id} caught={caughtPokemon} favorite={this.catchPokemon} />
-        }}/>
-        </Switch>
+        <AuthProvider>
+          <Switch>
+            <Route path="/signup" component={Signup} />
+            <Route path="/login" component={Login} />
+            <PrivateRoute
```

Comment:
I really like the use of this Private Route component only allowing the user to see things once they are logged in!